pkg/clusterapi/apibuilder.go

```diff
@@ -298,3 +280,36 @@ func MachineDeployment(clusterSpec *cluster.Spec, workerNodeGroupConfig anywhere
 
 	return *md
 }
+
+// EtcdadmCluster builds a etcdadmCluster based on an eks-a cluster spec and infrastructureObject.
+func EtcdadmCluster(clusterSpec *cluster.Spec, infrastructureObject APIObject) *etcdv1.EtcdadmCluster {
```

Comment:
nit
```suggestion
func EtcdadmCluster(clusterSpec *cluster.Spec, machineTemplate APIObject) *etcdv1.EtcdadmCluster {
```

Reply:
why? it is referencing to a `InfrastructureTemplate` and i dont see why not naming it `infrastructureObject`.

Reply:
IMO `infrastructureObject` is less clear. What does object mean in this context, a kubernetes object? But then what is an infrastructure kubernetes object?

If you prefer to use `infrastructure` since that's how etcdadmcluster names the field, then I think `infrastructureTemplate` is better than `object`.